internal/controller/dnsrecord_controller.go

```diff
@@ -194,26 +207,39 @@ func (r *DNSRecordReconciler) publishRecord(ctx context.Context, dnsRecord *v1al
 	}
 	err := r.Get(ctx, client.ObjectKeyFromObject(managedZone), managedZone, &client.GetOptions{})
 	if err != nil {
-		return err
+		return "0s", err
 	}
 	managedZoneReady := meta.IsStatusConditionTrue(managedZone.Status.Conditions, "Ready")
 
 	if !managedZoneReady {
-		return fmt.Errorf("the managed zone is not in a ready state : %s", managedZone.Name)
+		return "0s", fmt.Errorf("the managed zone is not in a ready state : %s", managedZone.Name)
 	}
 
-	if dnsRecord.Generation == dnsRecord.Status.ObservedGeneration {
-		logger.V(3).Info("Skipping managed zone to which the DNS dnsRecord is already published", "dnsRecord", dnsRecord.Name, "managedZone", managedZone.Name)
-		return nil
+	// cut off here for the short reconcile loop
+	requeueIn := validFor
+	if dnsRecord.Status.ValidFor != "" {
+		requeueIn, _ = time.ParseDuration(dnsRecord.Status.ValidFor)
```

Comment:
shoud we be ignoring the error here? Is it possible we have a bad value?

Reply:
possible only if someone goes into the CR and changes values manually. This value is being set by `.String()` call on the duration struct. In case someone messes up with it will be `0` Duration. Then the logic later on will look like this:
```
expiryTime := previousReconcileStart.Add(0)
       // new reconcile will always be After expiryTime, so this clause always ends up as false 
	if !generationChanged(dnsRecord) && reconcileStart.Before(&expiryTime) {
               // will not requeue 
		return 0, nil
	}
```
and we won't use the `requeueIn` anywhere else so the zero duration doesn't matter. 

In other words, we really don't care about an error there - user would need to manually play with it and of he does and sets an invalid value we will end up ignoring it